packages/web/src/features/languages/ImportLanguageGlossesView.tsx

```diff
@@ -0,0 +1,143 @@
+      console.error(error);
+      flash.error(t('import_glosses', { context: 'error' }));
+    }
+  }
+
+  const [isSubmitting, setIsSubmitting] = useState<boolean>(false);
+  const [showConfirmationDialog, setShowConfirmationDialog] =
+    useState<boolean>(false);
+
+  return (
+    <>
+      <View fitToScreen className="flex justify-center items-start">
+        <div className="mx-4 flex-shrink">
+          <ViewTitle className="flex">
+            <span>{language.data.name}</span>
+            <span className="mx-2">-</span>
+            <span>{language.data.code}</span>
+          </ViewTitle>
+          <div className="flex flex-col gap-4">
+            <div>
+              <Trans i18nKey="import_description">
+                Select a language to import glosses from
+                <a
+                  href="https://hebrewgreekbible.online"
+                  className="text-blue-600  focus:underline hover:underline"
+                >
+                  hebrewgreekbible.online
+                </a>
+                .
+              </Trans>
+            </div>
+            <Form context={formContext} onSubmit={onSubmit}>
+              <div className="mb-2">
+                <FormLabel htmlFor="import">
+                  {t('import_language').toUpperCase()}
+                </FormLabel>
+                <SelectInput
+                  id="import"
+                  name="import"
+                  className="w-full"
+                  autoComplete="off"
+                  required
+                  aria-describedby="import-error"
+                  disabled={isSubmitting}
+                >
+                  {importLanguages.data.map((name) => (
+                    <option value={name} key={name}>
+                      {name}
+                    </option>
+                  ))}
+                </SelectInput>
+                <InputError id="import-error" name="import" context="import" />
+              </div>
+              {isSubmitting ? (
+                <div>
+                  <LoadingSpinner className="inline-block" />
+                  <span role="status" className="sr-only">
+                    {t('submitting')}
+                  </span>
+                </div>
+              ) : (
+                <Button onClick={() => setShowConfirmationDialog(true)}>
+                  <Icon icon="file-import" className="me-4"></Icon>
+                  {t('import_glosses')}
+                </Button>
+              )}
+            </Form>
+          </div>
+        </div>
+      </View>
+      <ConfirmationDialog
+        title={t('import_glosses')}
+        description={t('import_glosses_confirmation_warning')}
+        confirmationValue={formContext.getValues()['import']}
+        isOpen={showConfirmationDialog}
+        setIsOpen={setShowConfirmationDialog}
+        onConfirm={async () => {
+          setIsSubmitting(true);
+          await onSubmit(formContext.getValues());
+          setIsSubmitting(false);
```

Comment:
Hmm. At my normal job, we have a way of opening a modal in the submit handler and then awaiting until it is closed before continuing with the submit. I really like how simple that pattern is. Something like:
```ts
async function onSubmit() {
  const { status } = await modal.show()
  if (status === 'ok') {
    // finish submitting
  }
}
```

There is already a SubmittingIndicator that we can use while the form is submitting and the submit button is disabled while submitting as well, so it would be nice if we could use this pattern.

If you like this pattern, we could accomplish this by using forwardRef on the dialog and providing a function `show` that returns a promise for when the modal is closed and returns whether the modal was canceled or confirmed. Then we'd have to put a ref on the component so that we can open it programmatically

Reply:
Ok! I wanted to use submitting indicator but I couldn't since I'm not using normal form submission.

Reply:
I think I know what you're asking for, but I got stuck in the implementation: https://github.com/arrocke/gloss-translation/pull/123#discussion_r1269212125

I pushed my commits so you can take a look.

Reply:
Done